src/frontend/src/components/CheckList.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of FinishLine by NER and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import PageBlock from '../layouts/PageBlock';
+import { Form } from 'react-bootstrap';
+import styles from '../stylesheets/components/CheckList.module.css';
+import { ReactNode, useState } from 'react';
+
+export type CheckListItem = {
+  details: string;
+  resolved: boolean;
+};
+
+interface CheckListProps {
+  title: string;
+  headerRight?: ReactNode;
+  items: CheckListItem[];
+}
+
+const CheckList: React.FC<CheckListProps> = ({ title, headerRight, items }) => {
+  const [checks, setChecks] = useState(items);
+
+  const handleCheck = (idx: number) => {
+    const updatedChecks = checks.map((check, i) => {
```

Comment:
ok so what i meant in my previous review was that you don't need the map at all but i forgot we shouldn't mutate the list without setting it. technically this way still mutates the old list but it's fine becuase we set it right after i think.

so i think a better way to do this entire map thing (we'll see if we get errors lol) is:
`const updatedChecks = [...checks]; updatedChecks[idx].resolved = !updatedChecks[idx].resolved; setChecks(updatedChecks);`